### ISSUE TITLE
Follow up on The VMSS resource already exists - to be managed via Terraform this resource needs to be imported into the State 

### ISSUE THREAD

**Issue by zioalex** [hashicorp/terraform-provider-azurerm#17193]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.1.6

### AzureRM Provider Version

">= 2.97.0, ~> 2.99.0

### Affected Resource(s)/Data Source(s)

azurerm_linux_virtual_machine_scale_set

### Terraform Configuration Files

```hcl
# module.linuxVmss["MY_VMSS"].azurerm_linux_virtual_machine_scale_set.vmss["linux"] must be replaced
+/- resource "azurerm_linux_virtual_machine_scale_set" "vmss" {
      - encryption_at_host_enabled                        = false -> null
      - eviction_policy                                   = "Deallocate" -> null # forces replacement
      ~ id                                                = "/subscriptions/MY_SUBS/resourceGroups/dev-rg/providers/Microsoft.Compute/virtualMachineScaleSets/dev-vmss-ubuntu" -> (known after apply)
      ~ instances                                         = 4 -> 8
        name                                              = "MY_VMSS"
      ~ platform_fault_domain_count                       = 1 -> (known after apply)
      ~ priority                                          = "Spot" -> "Regular" # forces replacement
      - secure_boot_enabled                               = false -> null
      ~ tags                                              = {
          - "CreatedOnDate"                     = "2022-05-13T12:55:54.9487562Z" -> null
          - "Creator"                           = "The_creator" -> null
          - "Environment"                       = "DEV" -> null
          - "__AzureDevOpsElasticPool"          = "MY_POOL" -> null
          - "__AzureDevOpsElasticPoolTimeStamp" = "5/20/2022 1:57:59 PM" -> null
            # (5 unchanged elements hidden)
        }
      ~ unique_id                                         = "3e96771d-cb83-4736-b12a-0f1326d26826" -> (known after apply)
      - vtpm_enabled                                      = false -> null
        # (18 unchanged attributes hidden)
```

### Debug Output/Panic Output

```shell
module.linuxVmss["MY_VMSS"].azurerm_linux_virtual_machine_scale_set.vmss["linux"]: Creating...
╷
│ Error: A resource with the ID "/subscriptions/MY_SUBSC/resourceGroups/MY_RG/providers/Microsoft.Compute/virtualMachineScaleSets/MY_VMSS" already exists - to be managed via Terraform this resource needs to be imported into the State. Please see the resource documentation for "azurerm_linux_virtual_machine_scale_set" for more information.
│ 
│   with module.linuxVmss["MY_VMSS"].azurerm_linux_virtual_machine_scale_set.vmss["linux"],
│   on ../../local_tf_modules/my_modules/terraform/vmss_linux.tf line 8, in resource "azurerm_linux_virtual_machine_scale_set" "vmss":
│    8: resource "azurerm_linux_virtual_machine_scale_set" "vmss" {
```

### Expected Behaviour

The expected behavior is that the VMSS is replaced or the priority updated without any entire VMSS replacement.

### Actual Behaviour

Terraform breaks out complaining that the resource is not managed by itself and suggesting to re-import it.
This is very weird having Terraform recognized the change in Priority as a "forces replacement" update.

### Steps to Reproduce

Run my pipeline.

### References

https://github.com/hashicorp/terraform-provider-azurerm/issues/16910

**Comment by zioalex** [comment]
@tombuildsstuff  I am kind of disappointed for the quickness  the other ticket was closed ([#16910](https://github.com/hashicorp/terraform-provider-azurerm/issues/16910) without the possibility to explain further. 

Hope this time it will be better.
We are using a Storage Account to store the TF statefile and all resources are managed by Terraform. Nevertheless when same changes occurred to the scaleset Terraform doesn't recognize anymore that it created the resource.

**Comment by tombuildsstuff** [comment]
hi @zioalex 

As mentioned in #16910 unfortunately this is a usage question and not a bug in the Azure Provider so please [open an issue on the Community Discuss forum](https://github.com/hashicorp/terraform-provider-azurerm/issues/16910#issuecomment-1150024984) where someone should be able to assist further.

Thanks!
